processing/src/main/java/org/apache/carbondata/processing/surrogatekeysgenerator/csvbased/BadRecordslogger.java

```diff
@@ -89,41 +95,55 @@ public static String hasBadRecord(String key) {
     return badRecordEntry.remove(key);
   }
 
-  public void addBadRecordsToBilder(Object[] row, int size, String reason, String valueComparer) {
+  public void addBadRecordsToBuilder(Object[] row, int size, String reason, String valueComparer,
+      boolean badRecordsLogRedirect, boolean badRecordLoggerEnable) {
     StringBuilder logStrings = new StringBuilder();
+    size = row.length;
```

Comment:
better to validate whether we need to log the bad record in the starting of method, and then continue the log string creation logic